pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -345,39 +359,87 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACL(ctx context.Context, volumeID string, vm *vmoperatortypes.VirtualMachine, instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig) error {
+// configureFileVolumeACLWithDelete helps to remove net permissions for file volumes
+// Net permissions will be removed when the external facing IP is the only entry in
+// the ExternalIPtoClientVms list obtained from cnsfilevolumeclient instance
+// For other cases, the method will remove the vm entry from ExternalIPtoClientVms
+//  Returns error if any operation fails
+func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACLWithDelete(ctx context.Context, volumeID string, vm *vmoperatortypes.VirtualMachine, instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig) error {
 	log := logger.GetLogger(ctx)
-	networkProvider, err := cnsoperatorutil.GetNetworkProvider(ctx)
+	tkgVMIP, err := r.getExternalIP(ctx, vm)
 	if err != nil {
-		msg := fmt.Sprintf("Failed to identify the network provider. Error: %+v", err)
+		msg := fmt.Sprintf("Failed to get CNSFileVolumeClient instance. Error: %+v", err)
```

Comment:
Change the error message to something like 'failed to get external IP for the vm. Error: +%v'

Reply:
Done